Assignment6.Rmd

```diff
@@ -72,11 +72,52 @@ table(M2$certified, M2$predict2)
 
 ```
 
+The latter one has a lower error rate.
+In the first model, the accurate number is 2186, the latter one's accurate number is 5363.
+
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
+T1<-read.csv("student.record.csv",header= TRUE)
+T2<- read.csv("student.course.csv", header = TRUE)
+# Predicting certified from the other variables in the M1 data frame. Which variables should you use?
+
+# use the student record csv to predict their GPA,classification tree approach 1
+library(tidyverse)
+library(tidyr)
+library(dplyr)
+library(rpart)
+c.tree3 <- rpart(GPAO ~ ANONID + CATALOG_NBR + ANON_INSTR_ID + GRD_PTS_PER_UNIT + DIV + SUBJECT + TERM, method = "anova", data = T2)
+printcp(c.tree3)
+post(c.tree3,file = "tree3.ps", title = "course")
+
+
+na_vec <- which(!complete.cases(T1$HSGPA))
+T1 <- T1[-na_vec, ]
+
+c.tree4 <- prune(c.tree3, cp = 0.012353 )
+post(c.tree4, file = "tree4.ps", title = "course")
+T2$predict1 <- predict(c.tree3, T2)
+T2$predict2 <- predict(c.tree4, T2)
+table_mat3 <- table(T2$GPAO, T2$predict1)
+table_mat4 <- table(T2$GPAO, T2$predict2)
+
+accuracy3 <- sum(diag(table_mat3)) / sum(table_mat3)
+accuracy4 <- sum(diag(table_mat4)) / sum(table_mat4)
+# means that model 2 is better than model 1.
+
+# another approach of classification tree, select needed variables
+D1 <- inner_join(T1,T2)
+D1 <- D1[!(is.na(D1$MAJOR1_DEPT) | D1$MAJOR1_DEPT==""| D1$MAJOR2_DEPT==""| D1$MAJOR3_DEPT==""), ]
+c.tree5 <- rpart(GPAO ~ HSGPA + SEX + MAJOR1_DEPT, method = "anova", data = D1)
+printcp(c.tree5)
+post(c.tree5, file = "tree5.ps", title = "MOOC")
+
+D1$predict3 <- predict(c.tree5, D1)
+table_mat5 <- table(D1$GPAO, D1$predict3)
+accuracy5 <- sum(diag(table_mat5)) / sum(table_mat5)
 
```

Comment:
Great job overall but include some explanation / write-up on which model is better.